Sources/MapboxNavigation/NavigationView.swift

```diff
@@ -66,20 +66,18 @@ open class NavigationView: UIView {
         navigationMapView.delegate = delegate
     }
     
-    var tileStoreLocation: TileStoreConfiguration.Location? = .default
-    private var _navigationMapView: NavigationMapView? = nil
-    lazy var navigationMapView: NavigationMapView = {
-        _navigationMapView?.frame = bounds
-        let navigationMapView = _navigationMapView ?? NavigationMapView(frame: bounds, tileStoreLocation: tileStoreLocation)
-        navigationMapView.isHidden = false
-        navigationMapView.translatesAutoresizingMaskIntoConstraints = false
-        
-        navigationMapView.delegate = delegate
-        navigationMapView.navigationCamera.viewportDataSource = NavigationViewportDataSource(navigationMapView.mapView,
-                                                                                             viewportDataSourceType: .active)
-        
-        return navigationMapView
-    }()
+    // :nodoc:
+    public var navigationMapView: NavigationMapView {
+        didSet {
+            oldValue.removeFromSuperview()
```

Comment:
Is there any low-hanging fruit for partially implementing #3576 here without being able to clone the map view?

Reply:
This happens in `PresentationAnimator` and `DismissalAnimator`. Since this is `NavigationView` property we cannot make assumptions for the user here (e.g. switch camera to different state or change puck type). So idea is to simply replace `NavigationMapView` with new object.